modules/renter/contractor/journal.go

```diff
@@ -105,10 +107,18 @@ func (j *journal) checkpoint(data contractorPersist) error {
 	if err := tmp.Close(); err != nil {
 		return err
 	}
-	if err := j.f.Close(); err != nil {
-		return err
+
+	// If rename or close failed in a previous call, fd will be invalid,
+	//	so don't try to close
+	if fd := j.f.Fd(); fd != uintptr(syscall.InvalidHandle) {
+		if err := j.f.Close(); err != nil {
+			return err
+		}
 	}
-	if err := os.Rename(tmp.Name(), j.filename); err != nil {
+
+	// Attempt rename up to 5 times to prevent issues with anti-virus software
+	err = build.Retry(5, 1000*Millisecond, func() error { return os.Rename(tmp.Name(), j.filename) })
```

Comment:
1 second might be excessive, 100ms should be ok. Also, prefer to put the function body on its own line

Reply:
Sure. 5 retries with 1 second delay is just my goto on Windows. It was working with no delay, so should be fine.

Reply:
```
err = build.Retry(5, 100*Millisecond, 
   func() error { return os.Rename(tmp.Name(), j.filename) })
```
or
```
err = build.Retry(5, 100*Millisecond, 
		func() error {
			 return os.Rename(tmp.Name(), j.filename) 
		})
```

Reply:
I think

```go
err = build.Retry(5, 100*Millisecond, func() error {
  return os.Rename(tmp.Name(), j.filename) 
})
```

is preferred